hpy/devel/include/cpython/hpyfunc_trampolines.h

```diff
@@ -69,4 +69,17 @@
         return _h2py(IMPL(_HPyGetContext(), _py2h(self), _py2h(obj), op)); \
     }
 
+/* With the cpython ABI, the implicit casts from Py_buffer* to HPy_buffer*
+ * happen to Just Work, which makes things easier. */
+#define _HPyFunc_TRAMPOLINE_HPyFunc_GETBUFFERPROC(SYM, IMPL) \
+    static int SYM(cpy_PyObject *arg0, HPy_buffer *arg1, int arg2) \
+    { \
+        return (IMPL(_HPyGetContext(), _py2h(arg0), arg1, arg2)); \
+    }
```

Comment:
I fear that casting `Py_Buffer*` to `HPy_Buffer*` is undefined behavior :(, although I'm not sure how to fix.
Also, I think that the signature of `SYM` should take a `Py_Buffer*`?
And probably it's worth putting an explicit case to `HPy_Buffer*` in the call to `IMPL`.

Reply:
Yes, it's UB, but I think it should work everywhere.
Good point about making the cast explicit.

Reply:
I think it's worth putting a comment explaining that we know it's UB